be/src/exprs/vectorized/time_functions.cpp

```diff
@@ -549,6 +549,97 @@ DEFINE_TIME_ADD_AND_SUB_FN(micros, TimeUnit::MICROSECOND);
 #undef DEFINE_TIME_SUB_FN
 #undef DEFINE_TIME_ADD_AND_SUB_FN
 
+Status TimeFunctions::datetime_floor_prepare(starrocks_udf::FunctionContext* context,
+                                             starrocks_udf::FunctionContext::FunctionStateScope scope) {
+    if (scope != FunctionContext::FRAGMENT_LOCAL) {
+        return Status::OK();
+    }
+
+    if (!context->is_notnull_constant_column(1) || !context->is_notnull_constant_column(2)) {
+        return Status::InternalError("datetime_floor just support const value");
```

Comment:
it seems datetime_floor(NULL, NULL, NULL) should return NULL; But will got a error

Reply:
> it seems datetime_floor(NULL, NULL, NULL) should return NULL; But we will get a error here.

`datetime_floor(NULL, NULL, NULL)` means `date_floor(NULL, NULL)`  function call in FE, and `date_floor` requires second parameter must be non-null, so it will return an erro in FE directly.

I removed this if code and add ut in FE. @stdpain